examples/platforms/utils/settings.cpp

```diff
@@ -385,6 +385,14 @@ ThreadError otPlatSettingsGet(otInstance *aInstance, uint16_t aKey, int aIndex,
                 index++;
             }
         }
+        else if ((block.key & OT_SETTINGS_KEY_ID_MASK) == (aKey & OT_SETTINGS_KEY_ID_MASK) &&
+                 (block.flag & kBlockDeleteFlag))
+        {
+            // delete the settings which have the same key id, but have different version
+            // TODO: we could also read this kind of settings, and provide them to upper layer for version migration
+            block.flag &= (~kBlockDeleteFlag);
+            utilsFlashWrite(address, reinterpret_cast<uint8_t *>(&block), sizeof(block));
+        }
```

Comment:
I don't think the `otPlatSettings*` API should be required to understand the encoding of the actual payload, because it requires every implementation to duplicate this same code. Additionally, I personally feel that the version should be handled at the higher layer, because delete isn't necesarrily what we want to do if the version changes. If we are going to go ahead and introduce a version field, we should support migration (optionally) for version changes.